src/catalogue.js

```diff
@@ -0,0 +1,119 @@
+import { first, pick, omit, get } from 'lodash';
+import * as xhr from './xhr';
+import Promise from 'bluebird';
+
+const BASIC_PROPS = ['title', 'summary', 'identifier', 'uri'];
+const REQUEST_DEFAULTS = {
+    types: ['attribute', 'metric', 'fact'],
+    paging: {
+        offset: 0,
+        limit: 'ALL'
+    }
+};
+
+function pickBasicProps(item) {
+    return pick(item.meta, BASIC_PROPS);
+}
+
+function deserializeFact(fact) {
+    return pickBasicProps(fact);
+}
+
+function deserializeMetric(metric) {
+    const basicProps = pickBasicProps(metric);
+    const metricProps = pick(metric.content, ['expression', 'format']);
+
+    return { ...basicProps, ...metricProps };
+}
+
+function deserializeAttribute(attribute) {
+    const basicProps = pickBasicProps(attribute);
+    const displayForm = first(attribute.content.displayForms);
+    const attributeProps = {
+        dimensionUri: get(attribute, ['content', 'dimension']),
+        granularity: get(attribute, ['meta', 'identifier']),
+        elementsUri: get(displayForm, ['links', 'elements']),
+        dfIdentifier: get(displayForm, ['meta', 'identifier']),
+        dfUri: get(displayForm, ['meta', 'uri']),
+        dateType: get(attribute, ['content', 'type'])
+    };
+
+    return { ...basicProps, ...attributeProps };
+}
+
+function deserializeDimension(dimension) {
+    const basicProps = pickBasicProps(dimension);
+    const attributes = get(dimension, ['content', 'attributes']);
+    const dimensionProps = {
+        attributes: attributes.map(deserializeAttribute)
+    };
+
+    return { ...basicProps, ...dimensionProps };
+}
+
+const DESERIALIZERS = {
+    fact: deserializeFact,
+    metric: deserializeMetric,
+    attribute: deserializeAttribute,
+    dimension: deserializeDimension
+};
+
+function deserializeItem(item) {
+    const type = first(Object.keys(item));
+    const props = DESERIALIZERS[type](item[type]);
+
+    return {
+        ...props,
+        type,
+        id: props.identifier,
+        isAvailable: true
+    };
+}
+
+function deserializeItems(items) {
+    return items.map(deserializeItem);
+}
+
+
+export function loadDateDimensions(projectId, options) {
+    const uri = `/gdc/internal/projects/${projectId}/loadDateDims`;
+    let request = {
+        ...REQUEST_DEFAULTS,
+        ...options
+    };
+
+    request = omit(request, ['filter', 'types', 'paging']);
+
+    const $promise = xhr.ajax(uri, {
+        type: 'POST',
+        data: { dateDimsRequest: request }
+    });
+
+    return Promise.resolve($promise).then(res => {
+        const { dimensions, unavailable } = res.dateDimsResponse;
+
+        return {
+            available: dimensions ? deserializeItems(dimensions) : [],
+            unavailable: unavailable ? deserializeItems(unavailable) : []
+        };
+    });
+}
+
+export function loadItems(projectId, options) {
```

Comment:
separate into two calls, one private, internal for API call that documents what the API needs in options, other external where we implement visualization metadata object to pass bucket information.